solana/restaking-v2/programs/restaking-v2/src/lib.rs

```diff
@@ -31,7 +37,10 @@ pub mod restaking_v2 {
         let common_state = &mut ctx.accounts.common_state;
 
         common_state.admin = ctx.accounts.admin.key();
-        common_state.whitelisted_tokens = whitelisted_tokens;
+        common_state.whitelisted_tokens = whitelisted_tokens
+            .iter()
+            .map(|token_mint| token_mint.into())
+            .collect::<Vec<StakeToken>>();
```

Comment:
You probably want `into_iter` here.

```suggestion
        common_state.whitelisted_tokens = whitelisted_tokens
            .into_iter()
            .map(StakeToken::from)
            .collect();
```